packages/smooth_app/lib/database/dao_product.dart

```diff
@@ -37,11 +43,23 @@ class DaoProduct extends AbstractSqlDao
     }
   }
 
+  /// Returns the [Product] that matches the [barcode], or null.
   Future<Product?> get(final String barcode) async {
-    final Map<String, Product> map = await getAll(<String>[barcode]);
-    return map[barcode];
+    final List<Map<String, dynamic>> queryResults =
+        await localDatabase.database.query(
+      _TABLE_PRODUCT,
+      columns: _columns,
+      where: '$_TABLE_PRODUCT_COLUMN_BARCODE = ?',
+      whereArgs: <Object>[barcode],
```

Comment:
barcode is always a String, don't know if it forces a list of type object but string would be better if possible

Reply:
Not convinced it would be better with an explicit `<String>[]` cast, as it would imply for code readers that we always need a `<String>[]` for `whereArgs`. Which is not the case, as we can put anything (which is one of the principal interests of this parameter, e.g. for `String` with quotes).
Ok, let's do it.